DVLSForLinux/install-dvls.ps1

```diff
@@ -348,19 +359,25 @@ New-DPSAdministrator -ConnectionString $Settings.ConnectionStrings.LocalSqlServe
 if ($DVLSVariables.DVLSCertificate) {
     Write-Host ("[{0}] Generating self-signed certificate" -F (Get-Date -Format "yyyyMMddHHmmss")) -ForegroundColor Green
 
-    & sudo $PwshExecutable -Command {
-        Param(
-            $DVLSVariables
-        )
+    $keyFile = "cert.key"
+    $keyTmpPath = Join-Path -Path $DVLSVariables.TmpFolder -ChildPath $keyFile
 
-        & openssl req -x509 -newkey rsa:4096 -sha256 -days 3650 -nodes -keyout cert.key -out cert.crt -subj ("/CN={0}" -F $DVLSVariables.DVLSHostName) -addext ("subjectAltName=IP:{0}" -F $DVLSVariables.DVLSHostName) > /dev/null 2>&1
-        & openssl pkcs12 -export -out cert.pfx -inkey cert.key -in cert.crt -passout pass: > /dev/null 2>&1
-    } -Args $DVLSVariables
+    $crtFile = "cert.crt"
+    $crtTmpPath = Join-Path -Path $DVLSVariables.TmpFolder -ChildPath $crtFile
+
+    $pfxFile = "cert.pfx"
+    $pfxTmpPath = Join-Path -Path $DVLSVariables.TmpFolder -ChildPath $pfxFile
+    $pfxDvlsPath = Join-Path -Path $DVLSVariables.DVLSPath -ChildPath $pfxFile
+
+    & openssl req -x509 -newkey rsa:4096 -sha256 -days 3650 -nodes -keyout $keyTmpPath -out $crtTmpPath -subj ("/CN={0}" -F $DVLSVariables.DVLSHostName) -addext ("subjectAltName=IP:{0}" -F $DVLSVariables.DVLSHostName) > /dev/null 2>&1
+    & openssl pkcs12 -export -out $pfxTmpPath -inkey $keyTmpPath -in $crtTmpPath -passout pass: > /dev/null 2>&1
+
+    & sudo Move-Item -Path $pfxTmpPath -Destination $pfxDvlsPath
```

Comment:
@alistek I reduced the elevated surface by using openssl with the current user and putting the temporary artifacts in a temporary directly. We then move the PFX as root as a second step.